packages/node-core/src/indexer/storeCache.service.ts

```diff
@@ -199,22 +348,62 @@ class CachedModel<
     });
   }
 
-  private async markPreviousHeightRecordsBatch(tx: Transaction): Promise<any[]> {
-    // Different with markAsDeleted, we only mark/close all the records less than current block height
-    // thus, and new record with current block height will not be impacted,
-    // advantage is this sql is safe to concurrency resolve with any insert sql
+  private getByFieldFromCache(field: keyof T, value: T[keyof T] | T[keyof T][], findOne?: boolean): T[] {
+    const joinedData: T[] = [];
+    const unifiedIds: string[] = [];
+    if (Object.keys(this.setCache).length !== 0) {
+      joinedData.concat(
+        Object.entries(this.setCache).map(([, model]) => {
+          if (model.isMatchData(field, value)) {
+            const latestData = model.getLatest().data;
+            unifiedIds.push(latestData.id);
+            return latestData;
+          }
+        })
+      );
+      // No need search further
+      if (findOne && joinedData.length !== 0) {
+        return joinedData;
+      }
```

Comment:
I think the getCache has to remain in sync with the setCache so we should just be able to use the get cache here?

Reply:
Yes, this make sense. getCache actually keep all latest record for each entity, search from getCache will be enough.
Also bulkCreate, bulkUpdate, set all sync with getCache.